src/pages/resultsView/ResultsViewPageStoreUtils.ts

```diff
@@ -396,4 +396,15 @@ export function doesQueryHaveCNSegmentData(
     } else {
         return _.some(detailedSamples, s=>!!s.copyNumberSegmentPresent);
     }
+}
+
+export function getLoadingMessage(numGenes:number, numSamples:number){
+    const numDataPoints = numGenes * numSamples;
+    if (numDataPoints < 1500) {
+        return "this shouldn't take long.";
+    } else if (numDataPoints < 50000) {
+        return "this can take several seconds.";
```

Comment:
this also depends on the network -- maybe we should make it vaguer?

Reply:
would it be worth doing some empirical data collection to see if we can draw an approximate curve, then we can say "this usually takes about X seconds"?

Reply:
I think use only really needs messaging fgor something we know will take a fair amount of time.  Like > 5 seconds.  That's why I kind of like just setting a timeout and turing on the message after the timeout.

Reply:
what should the message say?